langchain/retrievers/contextual_compression.py

```diff
@@ -0,0 +1,54 @@
+""""""
+from typing import List
+
+from pydantic import BaseModel, Extra
+
+from langchain.retrievers.document_filter.base import (
+    BaseDocumentFilter,
+    RetrievedDocument,
+)
+from langchain.schema import BaseRetriever, Document
+
+
+class ContextualCompressionRetriever(BaseRetriever, BaseModel):
+    """"""
+
+    base_filter: BaseDocumentFilter
+    """Filter for filtering documents."""
+
+    base_retriever: BaseRetriever
+    """Base Retriever to use for getting relevant documents."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+        arbitrary_types_allowed = True
+
+    def get_relevant_documents(self, query: str) -> List[Document]:
+        """Get documents relevant for a query.
+
+        Args:
+            query: string to find relevant documents for
+
+        Returns:
+            Sequence of relevant documents
+        """
+        docs = self.base_retriever.get_relevant_documents(query)
+        retrieved_docs = [RetrievedDocument.from_document(doc) for doc in docs]
```

Comment:
hm if we just use retrieved document here only seems a bit weird

it would be nice if normal retrievers could also return RetrievedDocuments

could check if `doc` already retrieved doc, convert if not. and just return retrieved doc?

retrieved doc may be a concept worth living in general schema?

Reply:
actually.... if we arent 100% sure about retrieved doc as an abstaction, could be kinda nice to have internal only? in whcih case maybe make it `_RetrievedDocument`?

Reply:
personally not super convinced it's a long term abstraction, so happy to make it `_RetrievedDocument`